release/models/optical-transport/openconfig-optical-attenuator.yang

```diff
@@ -112,6 +124,27 @@ module openconfig-optical-attenuator {
          maintain constant output power.  When false, the attenuator is set
          max attenuation or blocked.";
     }
+
+    leaf max-output-power {
+      type decimal64 {
+        fraction-digits 2;
+      }
+      units dBm;
+      description
+        "The max power level allowed on the output of attenuator. This leaf
+        is optional when in SYSTEM_CONTROLLED mode.";
+    }
+
+    leaf max-output-power-threshold {
+      type decimal64 {
+        fraction-digits 2;
+      }
+      units dB;
+      description
+        "If the system-derived-target-output-power is equal to or greater than
+        'max-output-power + max-output-power-threshold', a device alarm will
+        be raised. This leaf is only relevant when in SYSTEM_CONTROLLED mode.";
```

Comment:
Naive question here on optical alarms and OC, but which alarm do we expect to be raised?  Do we define it and refer to the appropriate alarm in OpenConfig?

Reply:
No problem. All of the optical vendors today use the generic alarms list in OC where the "text" leaf is just a vendor chosen strings. We haven't really gone down the path of trying to standardize and define every alarm. When we introduced the alarms list we discussed how that would be ideal, but given there can be 1000 alarms on a platform, it just hasn't been something there's been much progress to date on.

Reply:
ok, maybe it makes sense then to just refer to the oc tree for alarms in the description.  (to emphasize we expect the alarm to be set in [/system/alarms](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-system.html#system-alarms)

Reply:
Makes sense and added, thanks.